Sync/Synchronizers/Bookmarks/BookmarksSynchronizer.swift

```diff
@@ -59,23 +59,22 @@ class TrivialBookmarkStorer: BookmarkStorer {
         var success: [GUID] = []
         var failed: [GUID: String] = [:]
 
-        func onUpload(result: POSTResult) -> DeferredTimestamp {
-            modified = result.modified
+        func onUpload(result: POSTResult, lastModified: Timestamp?) -> DeferredTimestamp {
             success.appendContentsOf(result.success)
             result.failed.forEach { guid, message in
                 failed[guid] = message
             }
 
-            log.debug("Uploaded records got timestamp \(modified).")
-            local.setModifiedTime(modified, guids: result.success)
-            return deferMaybe(result.modified)
+            log.debug("Uploaded records got timestamp \(lastModified).")
+            local.setModifiedTime(lastModified ?? modified, guids: result.success)
```

Comment:
I guess we're using 0 as a default timestamp in case the `X-Weave-Timestamp` property isn't in the request. We can always do that for the `XLM` header.